IPython/terminal/pt_inputhooks/qt.py

```diff
@@ -1,10 +1,15 @@
 import sys
 from IPython.external.qt_for_kernel import QtCore, QtGui
 
+# If we create a QApplication, keep a reference to it so that it doesn't get
+# garbage collected.
+_appref = None
```

Comment:
At least one of the options we support seemingly doesn't - I tried without it and things went wrong. I can't remember which one I was testing with, though.